web/source/keyboards/kmwkeyboards.ts

```diff
@@ -1042,10 +1057,36 @@ namespace com.keyman.keyboards {
      **/
     register(x) {
       var options=x['options'];
+      // Represents the `resolve` function for the corresponding Promise built in `keymanCloudRequest`.
+      var sourcePromise: () => void;
+      const promiseid = x['timerid'];
+
+      let signalSuccess = function() {};
+      let signalFailure = function(err: string|Error) {};
+      const _this = this;
+      
+      if(promiseid) {
+        const promiseFuncs =  this.registrationResolvers[promiseid];
```

Comment:
Can we check that `promiseFuncs` is valid here?
```suggestion
        const promiseFuncs = this.registrationResolvers[promiseid];
```